src/io/RDF.h

```diff
@@ -86,6 +89,21 @@ class RDF : public PluginBase {
 	//! count the number of molecules per component
 	//! @todo: remove it and replace it by component.getNumMolecules()
 	void accumulateNumberOfMolecules(std::vector<Component>& components);
+	
+	void observeARDFMolecule(double dd, double cosPhi, double cosPhiReverse, unsigned cid1, unsigned cid2) {
+		
+		if(dd > _maxDistanceSquare) { return; }
+		size_t distanceBinID = floor( sqrt(dd) / binwidth() );
+		size_t angularBinID = floor( (-cosPhi + 1.)/ angularbinwidth() );
+		size_t angularBinIDReverse = floor((-cosPhiReverse + 1.)/ angularbinwidth() );
+		size_t binID = distanceBinID * _angularBins + angularBinID;
+		size_t binIDReverse = distanceBinID * _angularBins + angularBinIDReverse;
+		#if defined _OPENMP
+		#pragma omp atomic
```

Comment:
this omp atomic will only work for the next line, so you will need the same for the next line as well.

Reply:
So this addresses the ''one atomic operation missing'' part I am guessing? 

Results look good. If no further comments from your side, then this is done :-)

Reply:
yup looks good.